Source/Core/DiscIO/DirectoryBlob.h

```diff
@@ -171,9 +174,10 @@ class DiscContentContainer
 {
 public:
   template <typename T>
-  void AddReference(u64 offset, const std::vector<T>& vector)
+  void Add(u64 offset, std::vector<T> vector)
   {
-    return Add(offset, vector.size() * sizeof(T), reinterpret_cast<const u8*>(vector.data()));
+    return Add(offset, vector.size() * sizeof(T),
+               std::make_shared<std::vector<u8>>(std::move(vector)));
```

Comment:
I didn't realize this earlier, but because of how I suggested that you change this function, you can no longer call it for `T` other than `std::vector<u8>`. Since this compiles, I assume we're not doing that anywhere, so I would suggest making this function no longer templated.